windowing.py

```diff
@@ -0,0 +1,127 @@
+import pickle
+import numpy as np
+from skimage.util.shape import view_as_windows
+import statistics
+
+
+def create_windows_from_dictmatrix(dict_matrix, rows, window_size, stride):
+    # Description: takes dictionary that contains x, y, z, nanos keys and transforms
+    #			  this into an np array. Then creates windows from this matrix and returns
+    #			  the windows
+    #
+    # Input: dict_matrix = dict
+    # Input: rows = int (number of rows in a trace)
+    # Input: window_size = int (# of samples within one window when creating the windows)
+    # Input: stride = int (# of samples between the start of each window)
+    #
+    # Return: windowed_data = np array (np array of the windowed data)
+    x = dict_matrix['x']
+    y = dict_matrix['y']
+    z = dict_matrix['z']
+    nanos = dict_matrix['nanos']
+    np_data = np.array([x, y, z, nanos], 'd')
+    windowed_data = view_as_windows(np_data, (rows, window_size), step=stride)
+    return windowed_data
+
+
+def find_initial_lift_times(lift_windows, rows):
+    # Description: returns the start and end times from lift_windows np array
+    #
+    # Input: lift_windows = np array
+    # Input: rows = int (number of rows in a trace)
+    #
+    # Returns: int start (start time in nanos)
+    #		   int end (end time in nanos)
+
+    start = lift_windows[0][rows-1][0]
+    end = lift_windows[-1][-1][-1]
+    return start, end
+
+
+def find_lift_windows(windowed_data, divisor):
+    # Description: takes the windowed data and calculates variance for each window.
+    #				Then calculates variance threshold and returns an np array of
+    #				the windows that meet the variance threshold.
+    #
+    # Input: windowed_data = np array
+    # Input: divisor = int (will use divisor to calculate variance threshold so
+    #						that the threshold is a function of the max variance.
+    #						Threshold = (max variance)/divisor)
+    #
+    # Return: lift_windows = np array
+
+    windows = windowed_data[0]
+    windows_no_nanos = np.delete(windows, 3, 1)
+    variances = np.var(windows_no_nanos, axis=2, ddof=1)
+    var_sum = np.sum(variances, axis=1)
+    indices = np.argwhere(var_sum > max(var_sum)/6)
+    if (len(indices) <= 1):
+        indices = indices[0]
+    else:
+        indices = np.squeeze(indices)
```

Comment:
Could you remove this for loop and do array operations to check if the elements increment by 1? I think numpy.diff actually does this.